index.bs

```diff
@@ -1455,21 +1455,23 @@ When an [=XR input source=] for {{XRSession}} |session| begins its [=primary act
 
 <div class="algorithm" data-algorithm="on-input-end">
 
-When an [=XR input source=] for {{XRSession}} |session| ends its [=primary action=] the UA MUST run the following steps:
+When an [=XR input source=] |source| for {{XRSession}} |session| ends its [=primary action=] the UA MUST run the following steps:
 
+  1. Let |frame| be a new {{XRFrame}} with {{XRFrame/session}} |session| for the time the event occurred.
```

Comment:
It's a bit tedious to create the frame separately each time. I could also create the frame in the algorithm, however we may need to more strongly imply that the frame refers to poses from the time of the event, given that the algorithm is always called within a queued task (and thus the frame should not refer to poses from the time of the _task_)

Reply:
Yeah, I can see that this is a bit of a pain, but I think your logic here is sound and that this captures the intended timing well.